RadClass/H0.py

```diff
@@ -69,21 +90,38 @@ def run_gross(self, data, timestamp):
             self.x1_timestamp = timestamp
 
     def run_channels(self, data, timestamp):
+        '''
+        Applies scipy.stats.binomtest (requires v. 1.7.0 or greater)
+        to the channel/energy-wise count-rate. Spectral data is
+        passed by RadClass.RadClass and binomtest is applied to
+        each channel individually. Count-rates, x1 and x2, for
+        every spectral bin are tracked with associated timestamp
+        for next binomtest at next timestamp.
+        '''
         # only needed for the first initialization
         if self.x1 is None:
             self.x1 = data
             self.x1_timestamp = timestamp
         else:
             self.x2 = data
-            rejections = np.ones_like(data)
+            rejections = np.zeros_like(data)
             nvec = self.x1 + self.x2
             p = 0.5
             for i, (x1, n) in enumerate(zip(self.x1, nvec)):
-                pval = stats.binom_test(x1, n, p,
-                                        alternative='two-sided')
-                if pval <= self.significance:
+                # scipy.stats.binomtest will fail if n (# of trials)
+                # is less than 1 (possible for high-energy bins)
+                if int(n) < 1:
+                    pval = 0.0
+                else:
+                    pval = np.log10(stats.binomtest(int(x1), int(n), p,
+                                                    alternative='two-sided').pvalue)
+                    if np.isinf(pval):
+                        pval = -350.0  # np.log10(1E-350)
```

Comment:
This extra `if` is needed because sometimes `stats.binomtest.pvalue` can be so small that it gets rounded to `0.0` itself, meaning that `np.log10(0.0) = -inf`, which needs to be set to something, otherwise it will be saved as `0.0` like the default. This may not be the best way around this, so I'm open to suggestions/comments.